packages/ui-plugin-sheets/src/controllers/status-bar.controller.ts

```diff
@@ -0,0 +1,87 @@
+import { BaseValueObject, FormulaEngineService, ISheetData } from '@univerjs/base-formula-engine';
+import { RangeReferenceObject } from '@univerjs/base-formula-engine/reference-object/range-reference-object.js';
+import { ISelectionWithStyle, SelectionManagerService } from '@univerjs/base-sheets';
+import { Disposable, IRange, IUniverInstanceService, LifecycleStages, ObjectMatrix, OnLifecycle } from '@univerjs/core';
+import { Inject } from '@wendellhu/redi';
+
+import { IStatusBarService, IStatusBarServiceStatus } from '../services/status-bar.service';
+
+@OnLifecycle(LifecycleStages.Ready, StatusBarController)
+export class StatusBarController extends Disposable {
+    constructor(
+        @IUniverInstanceService private readonly _univerInstanceService: IUniverInstanceService,
+        @Inject(SelectionManagerService) private readonly _selectionManagerService: SelectionManagerService,
+        @Inject(FormulaEngineService) private readonly _formulaEngineService: FormulaEngineService,
+        @IStatusBarService private readonly _statusBarService: IStatusBarService
+    ) {
+        super();
+
+        this._init();
+    }
+
+    private _init(): void {
+        this._registerSelectionListener();
+        this._listenStatusChange();
+    }
+
+    private _registerSelectionListener(): void {
+        this._selectionManagerService.selectionInfo$.subscribe((selectionRangeWithStyle) => {
+            if (selectionRangeWithStyle) {
+                this._calculateSelection(selectionRangeWithStyle);
+            }
+        });
+    }
+
+    private _calculateSelection(selections: ISelectionWithStyle[]) {
+        const unitId = this._univerInstanceService.getCurrentUniverSheetInstance().getUnitId();
```

Comment:
There are redundant calls to `this._univerInstanceService.getCurrentUniverSheetInstance()`. You only have to call it once.